### PR TITLE
add upx-v4.0.2: new package

### PR DESCRIPTION
**Pull request by goproslowyo**
Related: https://github.com/wolfi-dev/os/issues/2921

#### For new package PRs only
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [x] REQUIRED - The package is added to `packages.txt`

I am opening this PR to add upx. Everything seems to build and package cleanly but I did notice that the program crashed when I tried to run it. Would appreciate some eyes on that to debug and push it over the line.

### CODE REVIEW

File: upx.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  description: "The Ultimate Packer for eXecutables"
+  copyright:
+    - license: GPL-2.0
+
+environment:
+  contents:
+    # repositories:
+    #   - https://packages.wolfi.dev/os
+    # keyring:
+    #   - https://packages.wolfi.dev/os/wolfi-signing.rsa.pub
```
Comment (rawlingsj) [inline upx.yaml:14]: We can remove these comments, when building locally we can run
```
USE_CACHE=no MELANGE=melange make packages/upx  BUILDWORLD=no
```
which adds the repository and keyring using CLI flags

**Review by rawlingsj — DISMISSED** [review]
This looks great @goproslowyo!  Just a couple of nit comments.

**Comment by rawlingsj** [comment]
It could also do with a rebase to avoid the "branch out of date" warning above.

**Comment by goproslowyo** [comment]
Thanks @rawlingsj  -- addressed those changes.

**Comment by goproslowyo** [comment]
Hi @rawlingsj or @jdolitsky or @deitch  or @kaniini --Following up! This has been ready for a few days and I keep rebasing it to keep up to date with main but no one seems to be reviewing it.

**Comment by dlorenc** [comment]
Sorry for the delay! Kicked off the tests.

**Comment by dlorenc** [comment]
Looks like a few build/lint errors.

**Comment by goproslowyo** [comment]
Great, taking a look, thanks!

**Comment by goproslowyo** [comment]
Thakns! Looks like the `env` key was invalid -- removed and hopefully we're going for green now!

**Comment by dlorenc** [comment]
Sorry this has been so painful - when you get your first PR merged the github actions won't need someone to approve anymore :(

**Comment by goproslowyo** [comment]
> Sorry this has been so painful - when you get your first PR merged the github actions won't need someone to approve anymore :(

All good! More painful was the CI -- I think maybe build `yam` into the devcontainer or make it a git pre-hook or something so I knew beforehand :)

**Review by dlorenc — APPROVED** [review]

**Comment by dlorenc** [comment]
Let's go!

**Comment by goproslowyo** [comment]
Awesome! Thanks everyone excited to have contributed and looking forward to
more hopefully!

**Comment by eddiezane** [comment]
Hey @goproslowyo,

Thanks for this contribution! I'd like to send some swag your way. If interested could you please shoot an email to eddiezane at chainguard dot dev?

**Pull request merged**